### PR TITLE
Add TypeScript definitions

### PR DESCRIPTION
**Pull request by forabi**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  type Options<T = Record<string, any>> = Partial<{
+    /** The name of the server entry point, defaults to 'main'. */
+    chunkName: string;
+
+    /** object Mixed in with clientStats & serverStats and passed to the serverRenderer. */
+    serverRendererOptions: T & {
+      clientStats: Stats;
+      serverStats: Stats;
+    };
+  }>;
```
Comment (richardscarrott) [inline index.d.ts]: Options is missing `createHandler` which I think should be of type: `(error: boolean, serverRenderer: Function) => Function`.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+declare module 'webpack-hot-server-middleware' {
+  import { RequestHandler } from 'express';
+  import { MultiCompiler, Stats } from 'webpack';
+  type Options<T = Record<string, any>> = Partial<{
+    /** The name of the server entry point, defaults to 'main'. */
+    chunkName: string;
+
+    /** object Mixed in with clientStats & serverStats and passed to the serverRenderer. */
+    serverRendererOptions: T & {
```
Comment (richardscarrott) [inline index.d.ts]: I'm not sure `options` should include `clientStats` and `serverStats` as you can't pass those into `webpackHotServerMiddleware`, they're mixed in with these options and passed into your `serverRenderer`.

File: index.d.ts (hunk 1 above)
Comment (forabi) [reply index.d.ts]: Is `createHandler` documented somewhere? I can't find it.

**Comment by forabi** [comment]
@richardscarrott I've updated the code

**Comment by richardscarrott** [comment]
@forabi Thanks for updating this -- I'll publish it on the weekend!

**Comment by forabi** [comment]
@richardscarrott Any news on this? Thanks in advance! 😄

**Comment by richardscarrott** [comment]
@forabi sorry for the slow response -- It's now available on npm @0.3.1

**Pull request merged**